# Concetti di ATE, ATT e ATU {#sec-ate-att-atu}

## Introduzione

Nel @sec-rct abbiamo visto come concettualizzare e calcolare l'ATE quando la randomizzazione è possibile, ovvero nel caso degli RCT. Tuttavia, non è sempre possibile o etico condurre esperimenti randomizzati e, in tali casi, abbiamo a disposizione solo i risultati di studi osservazonali. Tali studi, pur rivestendo un ruolo importante nella ricerca empirica, pongono notevoli sfide per l'inferenza causale, principalmente a causa del problema del confondimento. In questo capitolo, esploreremo la stima dell'effetto del trattamento in contesti osservazionali, dove il confondimento si manifesta quando variabili non controllate influenzano sia la probabilità di ricevere il trattamento sia l'outcome di interesse.

Per comprendere meglio come il confondimento possa distorcere le stime degli effetti causali, è utile esaminare come l'effetto di un trattamento (o intervento) possa variare all'interno di diverse popolazioni o gruppi. Tre concetti chiave in questo ambito sono l'**Effetto Medio del Trattamento** (Average Treatment Effect, **ATE**), l'**Effetto Medio del Trattamento sui Trattati** (Average Treatment Effect on the Treated, **ATT**), e l'**Effetto Medio del Trattamento sui Non Trattati** (Average Treatment Effect on the Untreated, **ATU**). Questi concetti sono fondamentali per comprendere come un intervento possa influenzare non solo l'intera popolazione, ma anche specifici sottogruppi al suo interno.

## Effetto Medio del Trattamento (ATE)

L'**ATE** rappresenta l'effetto medio che un trattamento ha su tutta la popolazione, indipendentemente dal fatto che gli individui abbiano effettivamente ricevuto il trattamento o meno. Formalmente, l'ATE è definito come la differenza media tra l'outcome che si otterrebbe se tutti fossero trattati e l'outcome che si otterrebbe se nessuno fosse trattato:

$$
\text{ATE} = E[Y^1 - Y^0]
$$

Dove $Y^1$ rappresenta l'outcome se un individuo riceve il trattamento e $Y^0$ rappresenta l'outcome se non lo riceve. L'ATE è utile per politiche che interessano l'intera popolazione, come un programma sanitario nazionale.

## Effetto Medio del Trattamento sui Trattati (ATT)

L'**ATT** misura l'effetto medio del trattamento solo per gli individui che hanno effettivamente ricevuto il trattamento. Questo concetto è particolarmente utile quando si vuole capire l'impatto del trattamento su coloro che lo hanno scelto o lo hanno ricevuto, ad esempio in uno studio osservazionale dove non tutti gli individui sono trattati:

$$
\text{ATT} = E[Y^1 - Y^0 \mid X = 1]
$$

Dove $X = 1$ indica che l'individuo ha ricevuto il trattamento. L'ATT è rilevante per valutare l'efficacia di un intervento sui partecipanti effettivi, come l'effetto di un programma di formazione sui partecipanti iscritti.

## Effetto Medio del Trattamento sui Non Trattati (ATU)

L'**ATU** misura l'effetto medio che il trattamento avrebbe avuto sugli individui che non hanno ricevuto il trattamento. È particolarmente utile per prevedere cosa accadrebbe se il trattamento fosse esteso a una popolazione che attualmente non lo riceve:

$$
\text{ATU} = E[Y^1 - Y^0 \mid X = 0]
$$

Dove $X = 0$ indica che l'individuo non ha ricevuto il trattamento. L'ATU è utile per valutare l'effetto potenziale dell'estensione di un trattamento a una popolazione che non è stata precedentemente trattata.

## Dati Ipotetici

Ecco una tabella di esempio che mostra i potenziali outcomes e gli effetti causali individuali (ICE) per un gruppo di individui:

| ID  | Età   | Trattato | Outcome con Trattamento (Y^1) | Outcome senza Trattamento (Y^0) | Effetto Causale Individuale (ICE, δ) | Outcome Osservato (Y) |
|-----|-------|----------|-------------------------------|---------------------------------|--------------------------------------|-----------------------|
| 1   | Anziano| 1        | 80                            | 60                              | 20                                   | 80                    |
| 2   | Anziano| 1        | 75                            | 70                              | 5                                    | 75                    |
| 3   | Anziano| 1        | 85                            | 80                              | 5                                    | 85                    |
| 4   | Anziano| 0        | 70                            | 60                              | 10                                   | 60                    |
| 5   | Giovane| 1        | 75                            | 70                              | 5                                    | 75                    |
| 6   | Giovane| 0        | 80                            | 80                              | 0                                    | 80                    |
| 7   | Giovane| 0        | 90                            | 100                             | -10                                  | 100                   |
| 8   | Giovane| 0        | 85                            | 80                              | 5                                    | 80                    |

## Calcolo di ATE, ATT e ATU

Possiamo calcolare i tre stimatori come segue:

- **ATE**: Media degli effetti causali individuali per tutti gli individui.
- **ATT**: Media degli effetti causali individuali solo per i trattati.
- **ATU**: Media degli effetti causali individuali solo per i non trattati.

Usando il seguente codice Python, possiamo calcolare questi valori dai dati della tabella.

In [1]:
import pandas as pd

# Creazione del dataframe
data = {
    "ID": [1, 2, 3, 4, 5, 6, 7, 8],
    "Età": [
        "Anziano",
        "Anziano",
        "Anziano",
        "Anziano",
        "Giovane",
        "Giovane",
        "Giovane",
        "Giovane",
    ],
    "Trattato": [1, 1, 1, 0, 1, 0, 0, 0],
    "Y1": [80, 75, 85, 70, 75, 80, 90, 85],
    "Y0": [60, 70, 80, 60, 70, 80, 100, 80],
}

df = pd.DataFrame(data)
df["ICE"] = df["Y1"] - df["Y0"]

# Calcolo di ATE, ATT, ATU
ATE = df["ICE"].mean()
ATT = df[df["Trattato"] == 1]["ICE"].mean()
ATU = df[df["Trattato"] == 0]["ICE"].mean()

ATE, ATT, ATU

(5.0, 8.75, 1.25)

Eseguendo il codice Python sopra, otteniamo:

- **ATE**: 5
- **ATT**: 8.75
- **ATU**: 1.25

Questi valori illustrano come l'effetto del trattamento vari a seconda del gruppo di interesse e offrono una comprensione più granulare dell'impatto di un intervento.

### La Differenza tra Dati Teorici ed Empirici nell'Inferenza Causale

I dati presentati nella tabella precedente rappresentano una situazione solo teorica, in cui possiamo osservare direttamente l'effetto che il trattamento avrebbe su ciascun individuo. Questo effetto è noto come **Effetto Causale Individuale** (Individual Causal Effect, **ICE**). In un tale mondo ipotetico, possiamo confrontare per ogni individuo l'outcome con e senza trattamento, calcolando esattamente quanto il trattamento abbia influenzato l'outcome di quell'individuo. Questo ci permette di calcolare l'**ATE** (Effetto Medio del Trattamento), l'**ATT** (Effetto Medio del Trattamento sui Trattati) e l'**ATU** (Effetto Medio del Trattamento sui Non Trattati) in modo preciso.

### La Natura Impossibile dei Dati Teorici

Tuttavia, questa situazione ideale è impossibile da realizzarsi empiricamente. Non possiamo osservare contemporaneamente l'outcome per un individuo sia con che senza trattamento. Questo perché, nella realtà, un individuo può trovarsi in una sola delle due condizioni: o riceve il trattamento o non lo riceve. Non possiamo tornare indietro nel tempo per osservare cosa sarebbe successo se un individuo che ha ricevuto il trattamento non l'avesse ricevuto, o viceversa. Questa impossibilità è alla base di ciò che viene chiamato il **problema fondamentale dell'inferenza causale**.

## Dati Empirici e il Problema della Confondimento

In un contesto empirico, i dati non sono costituiti da osservazioni teoriche come quelle della tabella precedente, ma piuttosto da due gruppi di individui: quelli che hanno ricevuto il trattamento e quelli che non l'hanno ricevuto. Non abbiamo accesso agli ICE per ciascun individuo; possiamo solo osservare i risultati degli individui che hanno ricevuto il trattamento e di quelli che non lo hanno ricevuto.

Un esempio di questi dati empirici è mostrato nella seguente tabella, dove vediamo solo il risultato osservato per ogni individuo e il fatto che essi abbiano o meno ricevuto il trattamento. Questo tipo di dati è comune negli studi osservazionali, dove il trattamento non è assegnato casualmente, ma piuttosto gli individui scelgono se partecipare o meno al trattamento, o sono selezionati per esso in base a determinati criteri.

| ID  | Età    | Trattato | Outcome Osservato (Y) |
|-----|--------|----------|-----------------------|
| 1   | Anziano| 1        | 80                    |
| 2   | Anziano| 1        | 75                    |
| 3   | Anziano| 1        | 85                    |
| 4   | Anziano| 0        | 60                    |
| 5   | Giovane| 1        | 75                    |
| 6   | Giovane| 0        | 80                    |
| 7   | Giovane| 0        | 100                   |
| 8   | Giovane| 0        | 80                    |

In questa tabella, l'**ID** identifica l'individuo, **Età** indica la fascia d'età dell'individuo (Anziano o Giovane), **Trattato** è una variabile binaria che indica se l'individuo ha ricevuto il trattamento (1 = sì, 0 = no), e **Outcome Osservato (Y)** è il risultato osservato per ciascun individuo. Questa tabella riflette ciò che sarebbe tipicamente disponibile in uno studio empirico, dove non possiamo osservare simultaneamente gli outcome con e senza trattamento per lo stesso individuo.

## Il Problema della Non Equivalenza dei Gruppi

Uno dei problemi principali negli studi osservazionali è che i due gruppi—quelli che ricevono il trattamento e quelli che non lo ricevono—non sono equivalenti. Questa non equivalenza è dovuta alla presenza di **variabili di confondimento**: fattori che influenzano sia la probabilità di ricevere il trattamento sia l'outcome di interesse. Ad esempio, in uno studio che esamina l'effetto di un programma di formazione sul reddito, l'età e il livello di istruzione potrebbero essere variabili di confondimento, poiché influenzano sia la probabilità di partecipare al programma sia il reddito.

## L'Impossibilità di Calcolare Diretamente l'ATE

A causa della presenza di queste variabili di confondimento, non possiamo semplicemente calcolare l'ATE come la differenza tra le medie dei due gruppi (trattati e non trattati). Se facessimo così, rischieremmo di ignorare l'influenza delle variabili di confondimento, ottenendo una stima distorta dell'effetto del trattamento. Ad esempio, se il gruppo trattato è composto da individui più giovani e istruiti rispetto al gruppo non trattato, la differenza osservata nei risultati potrebbe essere dovuta in parte a queste caratteristiche, piuttosto che all'effetto del trattamento stesso.

## Strategie per Superare il Problema del Confondimento

Per stimare correttamente l'**Effetto Medio del Trattamento** (ATE) o altri effetti del trattamento in studi osservazionali, è fondamentale affrontare il problema del confondimento. Il confondimento si verifica quando una variabile non osservata o non controllata influenza sia la probabilità di ricevere il trattamento sia l'outcome di interesse, rendendo difficile isolare l'effetto puro del trattamento. Esistono diverse tecniche statistiche sviluppate per tenere conto delle variabili di confondimento, ognuna delle quali offre un approccio diverso per correggere le distorsioni che il confondimento può introdurre.

### Matching

Il **matching** è una tecnica che mira a creare coppie o gruppi di individui trattati e non trattati che siano simili rispetto alle variabili di confondimento. L'idea alla base del matching è che, confrontando individui che sono simili in tutte le caratteristiche rilevanti tranne che per il trattamento ricevuto, si possa isolare l'effetto del trattamento stesso. Esistono diversi tipi di matching, tra cui:

- **Matching 1:1**: Ogni individuo trattato viene abbinato a un individuo non trattato che ha caratteristiche simili.
- **Matching caliper**: Gli individui sono abbinati solo se la distanza tra le loro variabili di confondimento è inferiore a una certa soglia.
- **Propensity Score Matching (PSM)**: Si calcola un punteggio di propensione (la probabilità di ricevere il trattamento data una serie di covariate) e si abbinano individui trattati e non trattati con punteggi di propensione simili.

Il matching riduce il bias di confondimento, ma richiede un campione abbastanza grande per trovare corrispondenze adeguate e può non eliminare completamente il confondimento, specialmente se alcune variabili importanti non sono osservate o misurate.

### Ponderazione Inversa della Probabilità (IPW)

La **ponderazione inversa della probabilità** (Inverse Probability Weighting, IPW) è un'altra tecnica per affrontare il confondimento, basata sull'assegnazione di pesi agli individui in base alla loro probabilità stimata di ricevere il trattamento. L'IPW utilizza modelli statistici per calcolare la probabilità (o punteggio di propensione) che un individuo riceva il trattamento dato un insieme di variabili di confondimento.

- Gli individui trattati che avevano una bassa probabilità di ricevere il trattamento ricevono un peso maggiore.
- Gli individui non trattati che avevano una bassa probabilità di non ricevere il trattamento ricevono un peso maggiore.

Questo metodo bilancia le distribuzioni delle variabili di confondimento tra i gruppi trattati e non trattati, in modo che i gruppi possano essere confrontati in modo più equo, come se fossero stati creati attraverso una randomizzazione.

Tuttavia, l'IPW richiede un modello accurato per stimare i punteggi di propensione e può essere sensibile ai pesi estremi, che possono amplificare le variazioni e introdurre instabilità nelle stime.

### Stratificazione

La **stratificazione** implica la divisione della popolazione in sottogruppi (o strati) omogenei rispetto a una o più variabili di confondimento. Una volta creati questi strati, si può calcolare l'effetto del trattamento all'interno di ciascun sottogruppo, riducendo il confondimento all'interno degli strati.

- Ad esempio, si potrebbe stratificare la popolazione per età, dividendo gli individui in fasce d'età, e poi calcolare l'effetto del trattamento all'interno di ciascuna fascia.

La stratificazione permette di controllare il confondimento senza la necessità di abbinare direttamente individui trattati e non trattati o di assegnare pesi. Tuttavia, la stratificazione può diventare complessa se ci sono molte variabili di confondimento, e il numero di individui in ciascun strato potrebbe diventare troppo piccolo per produrre stime affidabili.

### Altre Tecniche

Oltre a queste tecniche, esistono anche altre metodologie avanzate per affrontare il confondimento negli studi osservazionali:

- **Regressione con covariate**: L'inclusione delle variabili di confondimento come covariate in un modello di regressione permette di stimare l'effetto del trattamento al netto del confondimento.
- **Analisi delle variabili strumentali**: Si utilizza una variabile strumentale che è correlata con il trattamento ma non con l'outcome, tranne che attraverso il trattamento, per isolare l'effetto causale.
- **Propensity Score Weighting**: Una combinazione di matching e ponderazione basata sui punteggi di propensione per ottenere una distribuzione bilanciata delle covariate nei gruppi trattati e non trattati.

Queste tecniche, se applicate correttamente, permettono di isolare l'effetto causale del trattamento riducendo l'influenza delle variabili di confondimento. Tuttavia, è importante scegliere la tecnica appropriata in base alla natura dei dati e alle ipotesi sottostanti, nonché considerare la possibilità che il confondimento residuo possa ancora influenzare le stime. In ogni caso, queste metodologie avanzate migliorano significativamente la nostra capacità di fare inferenze causali accurate in studi osservazionali, avvicinandoci a stime più precise dell'ATE, ATT e ATU.

## Conclusione

In sintesi, mentre i dati teorici ci permettono di calcolare con precisione gli effetti causali individuali, i dati empirici raccolti in studi osservazionali presentano sfide significative a causa della non equivalenza dei gruppi e della presenza di variabili di confondimento. Per affrontare queste sfide, è essenziale utilizzare metodi statistici avanzati che permettano di stimare correttamente gli effetti del trattamento, evitando di commettere errori di interpretazione che potrebbero derivare da stime non corrette.